### ISSUE TITLE
What are the needed options for saving for offline use?

### ISSUE THREAD

**Issue by DeployThemAll** [gildas-lormeau/SingleFile#1292]
I'm trying to save a page for offline use using single-file.js injection into console

what are the options I need to add to download the full html page with css,js,uri images all included for offline use

singlefile.getPageData

**Comment by gildas-lormeau** [comment]
You can set the options as you prefer. In all cases, you should get a page viewable offline. Note however that you cannot process a full page from the console because of restrictions related to the [Same-origin policy](https://en.wikipedia.org/wiki/Same-origin_policy).

**Comment by DeployThemAll** [comment]
this also applies on injecting single-file.js using puppeteer?

**Comment by gildas-lormeau** [comment]
No, when using puppeteer, scripts can fetch resources from third-party domains without this restriction. If you want to play with SingleFile and puppeteer, take a look at https://github.com/gildas-lormeau/single-file-cli. It's already implemented.
